src/Lecture4.hs

```diff
@@ -181,7 +214,17 @@ instance for the 'Stats' type itself.
 -}
 
 instance Semigroup Stats where
-
+    a <> b = Stats {
+        statsTotalPositions = statsTotalPositions a <> statsTotalPositions b,
+        statsTotalSum       = statsTotalSum       a <> statsTotalSum       b,
+        statsAbsoluteMax    = statsAbsoluteMax    a <> statsAbsoluteMax    b,
+        statsAbsoluteMin    = statsAbsoluteMin    a <> statsAbsoluteMin    b,
+        statsSellMax        = statsSellMax        a <> statsSellMax        b,
+        statsSellMin        = statsSellMin        a <> statsSellMin        b,
+        statsBuyMax         = statsBuyMax         a <> statsBuyMax         b,
+        statsBuyMin         = statsBuyMin         a <> statsBuyMin         b,
```

Comment:
I think the existing implementation doesn't work on big file because here is lazy `Maybe` semigroup append and not strict 🤔